homeassistant/components/binary_sensor/urlwatch.py

```diff
@@ -0,0 +1,177 @@
+"""
+This component provides support for detecting changes to url content.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.urlwatch/
+"""
+import logging
+from datetime import timedelta
+import hashlib
+from os.path import exists
+from threading import Lock
+import pickle
+import voluptuous as vol
+import requests
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA)
+
+from homeassistant.const import (
+    CONF_NAME, CONF_RESOURCE, CONF_VERIFY_SSL)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['beautifulsoup4==4.6.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+DOMAIN = 'urlwatch'
+
+DEFAULT_NAME = 'URLWatch'
+DEFAULT_VERIFY_SSL = True
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_RESOURCE): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the URL watch sensor."""
+    name = config.get(CONF_NAME)
+    resource = config.get(CONF_RESOURCE)
+    method = 'GET'
+    verify_ssl = config.get(CONF_VERIFY_SSL)
+
+    data_file = hass.config.path("{}.pickle".format(DOMAIN))
+    storage = StoredData(data_file)
+
+    content = HttpData(method, resource, verify_ssl)
+    content.update()
+
+    if content.data is None:
+        _LOGGER.error("Unable to fetch data from %s", resource)
+        return False
+
+    add_devices([
+        URLWatchBinarySensor(hass, content, name, storage)
+    ])
+
+
+class URLWatchBinarySensor(BinarySensorDevice):
+    """Representation of a URL watch sensor."""
+
+    def __init__(self, hass, content, name, storage):
+        """Initialize a URL watch sensor."""
+        self.content = content
+        self._name = name
+        self._state = None
+        self._storage = storage
+        self.update()
```

Comment:
This means that you're fetching content twice at startup, you probably want to throttle the HttpData updater.